### ISSUE TITLE
Bad interactions between `total` and `macro` attributes

### ISSUE THREAD

**Issue by amelieled** [runtimeverification/k#3191]
### What component is the issue in?

Front-End

### Which command

- [X] kompile
- [ ] kast
- [ ] krun
- [ ] kprove
- [ ] kprovex
- [ ] ksearch

### What K Version?

K version:    v5.5.102

### Operating System

Linux

### K Definitions (If Possible)

```
module TEST-MACRO-SYNTAX

    syntax T ::= "A" | "B" // [macro]
    syntax D ::= "C"
    rule B => A
endmodule


module TEST-MACRO
    import TEST-MACRO-SYNTAX

    syntax D ::= "func" T [function, total]
    rule func A => C
    rule func B => C // (1)
endmodule
```

### Steps to Reproduce

The previous file compiles without warning.
However, if you uncomment the first comment, one can get the following error:
```
[Warning] Compiler: Non exhaustive match detected:
`func__TEST-MACRO_D_T`(`B_TEST-MACRO-SYNTAX_T`(.KList))
	Source(~/test-macro.k)
	Location(13,18,13,44)
	13 |	    syntax D ::= "func" T [function, total]
	   .	                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
```
I get exactly the same warning if I comment or not the line (1).

### Expected Results

No warning

**Comment by Baltoli** [comment]
@dwightguth My guess is that the fix for this will involve dropping macro productions before doing pattern-match compilation (and therefor exhaustiveness checking?)

We match macros syntactically [here](https://github.com/runtimeverification/llvm-backend/blob/master/tools/kore-expand-macros/main.cpp), so I think it should be sound to exclude them from the exhaustiveness check? I might be missing something though.

**Comment by dwightguth** [comment]
Yes, I would assume you should remove macros from the signature of pattern matching when doing exhaustiveness checks

**Comment by radumereuta** [comment]
We should drop the macro label somewhere in the scala code where we generate the LLVM code.
